hudi-common/src/main/java/org/apache/hudi/common/model/HoodieColumnRangeMetadata.java

```diff
@@ -30,16 +28,28 @@
   private final String columnName;
   private final T minValue;
   private final T maxValue;
-  private final long numNulls;
-  private final PrimitiveStringifier stringifier;
+  private long numNulls;
+  // For Decimal Type/Date Type, minValue/maxValue cannot represent it's original value.
+  // eg: when parquet collects column information, the decimal type is collected as int/binary type.
+  // so we cannot use minValue and maxValue directly, use minValueAsString/maxValueAsString instead.
+  private final String minValueAsString;
```

Comment:
Please take a look at my comment below -- i don't think this is necessary we can handle all type conversions at the time of reading of the Footer and don't need to propagate it further

Reply:
yes, I agree with you。But let's put this into #4060 as an optimization. 
We can merge the PR first and then merge the #4060

Reply:
sounds good to me as well.